### DISCUSSION TITLE
ScriptReference/AudioClip.GetData

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#293] [Notes]
# ScriptReference/AudioClip.GetData

https://docs.unity3d.com/ScriptReference/AudioClip.GetData.html

**Comment by joonjoonjoon** [comment]
this was kind of non obvious to me, but the sample count you get from a clip is for one channel.   
Meaning you'll have to multiply it by the amount of channels when preparing the array for GetData.  
It's in the example code (`new float[audioSource.clip.samples * audioSource.clip.channels];`) but I somehow missed it and took me a while to figure out.

**Comment by joethephish** [comment]
Sometimes I find (maybe only with ogg format?) that the number of samples available to GetData() is actually 1 sample less than would be calculated via `clip.samples * clip.channels`. I get the warning
> Data longer than the AudioClip: <clip name="">. 10434229 sample(s) copied

See how in that example the number of samples copied is an odd number, even though there are 2 channels.  
Also, the documentation seems to by lying about wrapping around? (thankfully!)
